streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -905,15 +913,19 @@ private long computeLatency() {
      */
     public synchronized void close() {
         log.info("{} Informed thread to shut down", logPrefix);
-        setState(State.PENDING_SHUTDOWN);
+        setState(State.PENDING_SHUTDOWN, true);
```

Comment:
Currently in testing we can call thread.close() several times (e.g., `testStateChangeStartClose`) E.g., when a thread is DEAD, one can still call thread.close(). This is probably not a production scenario, so not sure how important it is. But that's the reason.